cloudsmith_cli/cli/commands/upstream.py

```diff
@@ -0,0 +1,449 @@
+# -*- coding: utf-8 -*-
+"""CLI/Commands - create, retrieve, update or delete repository upstreams."""
+from __future__ import absolute_import, print_function, unicode_literals
+
+import json
+
+import click
+import six
+
+from ...core.api import upstreams as api
+from .. import command, decorators, utils, validators
+from ..exceptions import handle_api_exceptions
+from ..utils import (
+    fmt_bool,
+    fmt_datetime,
+    maybe_spinner,
+    maybe_truncate_list,
+    maybe_truncate_string,
+)
+from .main import main
+
+UPSTREAM_FORMATS = [
+    "dart",
+    "deb",
+    "docker",
+    "helm",
+    "maven",
+    "nuget",
+    "npm",
+    "python",
+    "rpm",
+    "ruby",
+]
+
+
+def print_upstreams(upstreams):
+    """Print upstreams as a table or output in another format."""
+
+    headers = [
+        "Name",
+        "Upstream Url",
+        "Auth mode",
+        "Auth Secret",
+        "Auth Username",
+        "Created At",
+        "Extra Header 1",
+        "Extra Header 2",
+        "Extra Value 1",
+        "Extra Value 2",
+        "Active",
+        "Mode",
+        "Priority",
+        "Slug Perm",
+        "Updated At",
+        "Verify SSL",
+        "Component",
+        "Distribution Versions",
+        "Upstream Distribution",
```

Comment:
Similar to the above, reckon we could hide these for formats that don't support distros?

Reply:
Will do, cheers.